src/reducer/database/models.ts

```diff
@@ -0,0 +1,1066 @@
+      repoUrl: fields.repoUrl,
+      datetime: fields.datetime,
+      nameWithOwner: fields.nameWithOwner,
+      type: fields.type,
+      calendarId: this.id,
+    });
+
+    return contribution as Contribution;
+  }
+
+  /**
+   * @todo Implement a save functionality
+   * @description Saves this osm model
+   */
+  save() {}
+}
+
+/**
+ * @class A OSM model for a contribution object
+ * @extends osm.models.ContributionSO Statement object
+ */
+class Contribution extends osm.models.ContributionSO implements IContribution {
+  public static objects = osm.models.ContributionSO.getObjects(Contribution);
+  public objects = Contribution.objects;
+
+  public id = 0;
+  public repoUrl = "";
+  public datetime = "";
+  public nameWithOwner = "";
+  public type = "";
+  public calendarId = 0;
+
+  constructor(args: IContribution) {
+    super();
+
+    this.id = args["id"];
+    this.repoUrl = args["repoUrl"];
+    this.datetime = args["datetime"];
+    this.nameWithOwner = args["nameWithOwner"];
+    this.type = args["type"];
+    this.calendarId = args["calendarId"];
+  }
+
+  /**
+   * @todo Implement a save functionality
+   * @description Saves this osm model
+   */
+  save() {}
+}
+
+/**
+ * @class A OSM model for a talk object
+ * @extends osm.models.TalkSO Statement object
+ */
+class Talk extends osm.models.TalkSO implements ITalk {
+  public static objects = osm.models.TalkSO.getObjects(Talk);
+  public objects = Talk.objects;
+
+  public id = 0;
+  public name = "";
+  public url = "";
+  public displayUrl = "";
+  public downloadUrl = "";
+  public path = "";
+  public repositoryData = "";
+
+  constructor(args: ITalk) {
+    super();
+
+    this.id = args["id"];
+    this.name = args["name"];
+    this.url = args["url"];
+    this.displayUrl = args["displayUrl"];
+    this.downloadUrl = args["downloadUrl"];
+    this.path = args["path"];
+    this.repositoryData = args["repositoryData"];
+  }
+
+  getRepository(): Repository {
+    try {
+      /* Parse repositoryData to a object */
+      let repositoryData = JSON.parse(this.repositoryData);
+
+      /* Filter repositories for repository represented by repositoryData */
+      const repository: any = Repository.objects.filter({
+        url: repositoryData.url,
+      })[0];
+
+      if (repository) {
+        repository.owner = Member.objects.get({
+          id: repository.ownerId,
+        }) as Member;
+
+        repository.owner.render([]);
+
+        /* Return the existing repository from the database */
+        return repository;
+      } else {
+        /* Return a new repository created with repositoryData */
+        let repository: any = new Repository(repositoryData);
+
+        repository.owner = new Member(repositoryData.owner);
+
+        return repository;
+      }
+    } catch (ex) {
+      console.warn(
+        ex +
+          "De-serialization of repositoryData failed, " +
+          "therefore an empty repository is provided!"
+      );
```

Comment:
Is the error anchor the right one for this case, if not please change the suggestion.
```suggestion
      //#ERROR
      console.warn(
        ex +
          "De-serialization of repositoryData failed, " +
          "therefore an empty repository is provided!"
      );
```

Reply:
Yes it is actually an error although it is issued by a console.warn.
But, there should a anchors for legacy and todo
```suggestion
      //#LEGACY
      //#TODO Implement proper error handling
      //#ERROR
      console.warn(
        ex +
          "De-serialization of repositoryData failed, " +
          "therefore an empty repository is provided!"
      );
```